### PR TITLE
Trim display name before applying it

### PR DESCRIPTION
**Pull request by Vlatombe**
The current implementation is not robust against blank display name

@reviewbybees

### CODE REVIEW

**Comment by rsandell** [comment]
:bee:

**Comment by oleg-nenashev** [comment]
:bee:

**Comment by KostyaSha** [comment]
Test cases?

File: test/src/test/groovy/hudson/model/AbstractBuildTest.groovy [hunk 1]
```diff
@@ -151,6 +151,24 @@ public class AbstractBuildTest {
...
+        p.setDisplayName(" ");
+        assertEquals("A blank display name should be ignored.", "foo", p.getDisplayName());
+    }
+
+    @Test void validDisplayName() {
+        FreeStyleProject p = j.createFreeStyleProject("foo");
+        p.setDisplayName("bar");
+        assertEquals("A non-blank display name should be used.", "bar", p.getDisplayName());
+    }
+
```
Comment (stephenc) [inline test/src/test/groovy/hudson/model/AbstractBuildTest.groovy]:
```
@Test void trimValidDisplayName() {
     FreeStyleProject p = j.createFreeStyleProject("foo");
     p.setDisplayName("    bar    ");
     assertEquals("A non-blank display name with whitespace should be trimmed.", "bar", p.getDisplayName());
}
```

**Comment by stephenc** [comment]
:ant: you missed one test case

**Comment by stephenc** [comment]
:bee:

**Comment by oleg-nenashev** [comment]
:bee:

**Comment by KostyaSha** [comment]
:+1:  with tests

**Comment by stephenc** [comment]
:bee:

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Comment by Vlatombe** [comment]
I'll squash then merge.

**Pull request merged**